### PR TITLE
feat(ui): add copy-to-clipboard for skill paths

### PR DESCRIPTION
**Pull request by salmonumbrella**
## Summary
- Adds a copy button (overlapping-squares icon) next to the "Path" label in the Metadata card on the skill detail page — copies the full absolute source path
- Adds the same copy button in the file viewer modal header next to the filename — copies the full file path
- Clicking shows a brief checkmark with "Copied!" confirmation

## Test plan
- [x] Open a skill detail page, click the copy icon next to "Path" — verify full absolute path is copied
- [x] Open a file in the file viewer modal, click the copy icon next to the filename — verify full file path is copied
- [x] Confirm "Copied!" feedback appears briefly after clicking

### CODE REVIEW

File: ui/src/pages/SkillDetailPage.tsx [hunk 1]
```diff
@@ -625,6 +633,26 @@ function MetaItem({
...
+        });
+      }}
+      className="inline-flex items-center gap-0.5 align-middle ml-1.5 text-pencil-light hover:text-pencil transition-colors cursor-pointer shrink-0"
+      style={{ position: 'relative', top: '1px' }}
+      title="Copy to clipboard"
+    >
+      {copied ? <><Check size={12} strokeWidth={2.5} /> <span className="text-xs">Copied!</span></> : <Copy size={12} strokeWidth={2.5} />}
+    </button>
+  );
+}
```
[the remark this answers is not in the document]
Comment (salmonumbrella) [reply ui/src/pages/SkillDetailPage.tsx]: Extracted the duplicated copy-to-clipboard logic into `ui/src/components/CopyButton.tsx` and switched both `SkillDetailPage` and `FileViewerModal` to the shared component. Added clipboard failure handling through the existing toast system, plus a Vitest regression test covering success, permission denial, and the success-then-failure edge case.

Verified with `npm test -- src/components/CopyButton.test.tsx` and `npm run build` in `ui/`.

**Comment by runkids** [comment]
Thanks for the contribution! Love the clean `CopyButton` component extraction — great reusability.

I'm about to merge this into main. Heads up that I've been doing a major UI redesign on the `v0.17.0` branch (touching `SkillDetailPage.tsx` and `FileViewerModal.tsx` among many others), so I'll handle the integration with the new UI during my rebase. Your `CopyButton.tsx` component itself won't need any changes — just the integration points.

Merging now 🎉

**Comment by salmonumbrella** [comment]
Thanks for merging @runkids. Tried the new UI. It's so corporate. I loved your original design—especially browsing the skills and the colors made it easy to read. Maybe you can find a happy medium between corporate and fun 🙏

**Comment by runkids** [comment]
@salmonumbrella Honestly, I'm with you—I actually preferred the previous version too! XD
It’s one of the toughest parts of being a builder: I received feedback that the original design was a bit too 'busy' for some workflows, so I pivoted to this more 'corporate' look. But now I see that the 'fun' side was what made it special for many.
Since there's no way to please everyone with a single design, I'm now exploring a 'Theme Switcher' or 'Skin' system so you can get that vibrant look back. Stay tuned, and thanks for being honest—it means a lot to me! 🙏

**Comment by salmonumbrella** [comment]
@runkids I am so glad you replied!! Skins are bloat. Your original design is wonderful. Not only was it fun to hover/click around, it made hundreds of skills feel approachable. Each skill felt like a sticky note pinned to my wall. 

The yellow colors you had were great. The formatting of the .md in the browser was lovely. Can't please everyone, but 0.17.0 reverted to looking like another B2B SaaS 🤣 Skillshare needs its personality back!! Thank you for building this 👊
